clang/lib/CIR/Dialect/Transforms/TargetLowering/Targets/NVPTX.cpp

```diff
@@ -63,6 +66,58 @@ class NVPTXTargetLoweringInfo : public TargetLoweringInfo {
 
 } // namespace
 
+bool NVPTXABIInfo::isScalarType(mlir::Type ty) const {
+  // OG checks if:
+  //     getKind() > BuiltinType::Void && getKind() <= BuiltinType::NullPtr
+  // which is confusing and unavailable in CIR.
+  // Here we manually list out all corresponding types in CIR.
+  return isa<IntType, BoolType, SingleType, DoubleType, FP16Type, FP128Type,
```

Comment:
Please add a helper function in `clang/include/clang/CIR/Dialect/IR/CIRTypes.h`, just like we have `isAnyFloatingPointType`, we should also have `isScalarType`, and just use it here.